### PR TITLE
Install the experimental CDN agent version

### PR DESCRIPTION
**Pull request by Finesse**

### CODE REVIEW

File: src/components/Layout/index.tsx [hunk 1]
```diff
@@ -92,8 +92,7 @@ export function LayoutTemplate({ children, siteMetadata, notificationBar }: Layo
         <script>
           {`(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0], j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src= 'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f); })(window,document,'script','dataLayer','${gtmToken}');`}
         </script>
-        <link rel='preconnect' href={FPJS_ENDPOINT} />
-        <link rel='preconnect' href={TLS_ENDPOINT} />
```
Comment (Finesse) [inline src/components/Layout/index.tsx:96]: The website code is not a source of truth for these endpoints anymore, so I've removed these preconnect tags

File: src/constants/env.ts [hunk 2]
```diff
@@ -13,6 +13,11 @@ export const FPJS_PUBLIC_TOKEN =
     [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_FPJS_PUBLIC_TOKEN,
   }) ?? 'test_public_token'
 
+export const FPJS_CDN_URL = getContextEnv<string | undefined>({
+  [NetlifyContext.Production]: process.env.GATSBY_FPJS_CDN_URL,
+  [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_FPJS_CDN_URL,
+})
```
Comment (Finesse) [inline src/constants/env.ts:19]: Please add the environment variables.

The production value is `https://c.fingerprintjs.com`.

The preview value depends on what environment the JS agent public key belongs to. If it belongs to the production, use `https://c.fingerprintjs.com` or `https://fpcdn.io`. If it belongs to the RC, use `https://procdn.fpjs.sh`.

File: src/constants/env.ts [hunk 3]
```diff
@@ -31,23 +36,12 @@ export const BOTD_TOKEN_ENDPOINT =
     [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_BOTD_TOKEN_ENDPOINT,
   }) ?? 'test_botd_token_endpoint'
 
-export const FPJS_ENDPOINT =
-  getContextEnv<string>({
-    [NetlifyContext.Production]: process.env.GATSBY_FPJS_ENDPOINT,
-    [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_FPJS_ENDPOINT,
-  }) ?? ''
```
Comment (Finesse) [inline src/constants/env.ts:38]: The variables may be removed, but I'd leave them just in case

File: src/constants/env.ts [hunk 4]
```diff
@@ -31,23 +36,12 @@ export const BOTD_TOKEN_ENDPOINT =
...
 export const FPJS_VISITORS_ENDPOINT =
   getContextEnv<string>({
     [NetlifyContext.Production]: process.env.GATSBY_FPJS_VISITORS_ENDPOINT,
     [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_FPJS_VISITORS_ENDPOINT,
   }) ?? 'https://history.demo.fpaux.net/pamplemousse/'
 
-export const TLS_ENDPOINT = getContextEnv<string>({
-  [NetlifyContext.Production]: process.env.GATSBY_TLS_ENDPOINT,
-  [NetlifyContext.DeployPreview]: process.env.GATSBY_PREVIEW_TLS_ENDPOINT,
-})
```
Comment (Finesse) [inline src/constants/env.ts:49]: The variables may be removed, but I'd leave them just in case

File: src/constants/env.ts [hunk 5]
```diff
@@ -86,8 +80,6 @@ export const BOTD_VERIFY_ENDPOINT =
 
 export const FPJS_REGION = process.env.GATSBY_FPJS_REGION
 export const FPJS_DASHBOARD_ENDPOINT = process.env.GATSBY_FPJS_DASHBOARD_ENDPOINT
-export const FPJS_MONITORING_CLIENT_ID = process.env.GATSBY_FPJS_MONITORING_CLIENT_ID
-export const FPJS_MONITORING_TOKEN = process.env.GATSBY_FPJS_MONITORING_TOKEN
```
Comment (Finesse) [inline src/constants/env.ts:90]: The variables may be removed, but I'd leave them just in case

File: src/context/FpjsContext.tsx [hunk 6]
```diff
@@ -33,29 +26,19 @@ export function FpjsProvider({ children }: { children: React.ReactNode }) {
...
+  const cdnUrl = FPJS_CDN_URL
 
-  const [visitorData, setVisitorData] = useState<ExtendedGetResult>()
+  const [visitorData, setVisitorData] = useState<FingerprintJS.ExtendedGetResult>()
   const [hasFpjsError, setHasFpjsError] = useState(false)
 
   useEffect(() => {
     async function getVisitorData() {
       try {
-        const FP = await import('@fingerprintjs/fingerprintjs-pro')
```
Comment (Finesse) [inline src/context/FpjsContext.tsx:48]: I don't know the exact reason for dynamic importing of JS agent. I suppose it's to reduce the weight of the pages that don't run the demo. The new agent version is a thin loader (it downloads the actual code from Pro CDN), so I've replaced the dynamic importing with a static importing.

If it is not an option, please let me know, I'll switch it back to the dynamic importing.

File: src/types/visitorResponse.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { ExtendedGetResult } from '@fingerprintjs/fingerprintjs-pro'
```
Comment (Finesse) [inline src/types/visitorResponse.ts:1]: Just a good practice to eliminate unwanted code in runtime (the tree-shaking can be not perfect)

File: src/types/visitorResponse.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
...
+import type { ExtendedGetResult } from '@fingerprintjs/fingerprintjs-pro'
```
Comment (Finesse) [inline src/types/visitorResponse.ts:1]: Just a good practice to eliminate unwanted code in runtime (the tree-shaking can be not perfect)

File: src/context/FpjsContext.tsx [hunk 9]
```diff
@@ -33,29 +26,19 @@ export function FpjsProvider({ children }: { children: React.ReactNode }) {
...
-        const fp = await FP.load({
+        const fp = await FingerprintJS.load({
           token: publicToken,
-          endpoint,
           region,
```
Comment (Finesse) [inline src/context/FpjsContext.tsx:40]: I've left the explicit region for a case when Pro CDN can't fetch data from MGMT API

**Comment by Finesse** [comment]
I have found no content security policy in the code or in the HTTP response headers on fingerprintjs.com. If there is a CSP somewhere, please whitelist the origins that you'll set to the FPJS_CDN_URL environment variables.

File: src/constants/env.ts (hunk 2 above)
Comment (natapg) [reply src/constants/env.ts:19]: I've added the following variables:
GATSBY_FPJS_CDN_URL: https://c.fingerprintjs.com
GATSBY_PREVIEW_FPJS_CDN_URL: https://procdn.fpjs.sh
and redeployed the deploy preview https://deploy-preview-528--fingerprintjscom.netlify.app/

Comment (Finesse) [reply src/constants/env.ts:19]: The deploy preview looks good 👍

**Review by natapg — APPROVED** [review]

**Comment by natapg** [comment]
> I have found no content security policy in the code or in the HTTP response headers on fingerprintjs.com. If there is a CSP somewhere, please whitelist the origins that you'll set to the FPJS_CDN_URL environment variables.

No, we don't have CSP exceptions on the website.

**Pull request merged**